pkg/controller/machine.go

```diff
@@ -199,6 +210,11 @@ func (c *controller) reconcileClusterMachine(machine *v1alpha1.Machine) error {
 	Machine controller - nodeToMachine
 */
 func (c *controller) addNodeToMachine(obj interface{}) {
+	if !cache.WaitForCacheSync(nil, c.nodeSynced) {
```

Comment:
Do we need this?

Reply:
As discussed offline, we can avoid it here I guess, thanks for bringing it up.